src/t_zset.c

```diff
@@ -3769,32 +3769,45 @@ void zrankGenericCommand(client *c, int reverse) {
     robj *key = c->argv[1];
     robj *ele = c->argv[2];
     robj *zobj;
+    robj* reply;
     long rank;
-    int opt_withscores = 0;
+    int opt_withscore = 0;
     double score;
 
-    if ((zobj = lookupKeyReadOrReply(c,key,shared.null[c->resp])) == NULL ||
-        checkType(c,zobj,OBJ_ZSET)) return;
-
-    serverAssertWithInfo(c,ele,sdsEncodedObject(ele));
-    if (c->argc > 3 && !strcasecmp(c->argv[3]->ptr, "withscores")) {
-        opt_withscores = 1;
+    if (c->argc > 4) {
+        sds err = sdsnew(NULL);
+        err = sdscatprintf(err, "wrong number of arguments for '%s' command", c->cmd->fullname);
```

Comment:
these can be replaced with `addReplyErrorArity`

Reply:
Oh you are right, I didn't see that function.